src/homework/sandro/FourInARow/FourInARow.java

```diff
@@ -0,0 +1,305 @@
+                if (field[i][j + 1] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (field[i][j + 2] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (field[i][j + 3] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (count == 4) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    public static boolean detectWinDiagonalfront1(int[][] field, int row, int col, int currentplayer) {
+        int count = 0;
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 3; j++) {
+                if (field[i][j] == currentplayer
+                        || field[i + 1][j] == currentplayer
+                        || field[i + 2][j] == currentplayer
+                        || field[i][j + 1] == currentplayer
+                        || field[i][j + 2] == currentplayer
+                        || field[i][j + 3] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (count == 4) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static boolean detectWinDiagonalback(int[][] field, int currentplayer) {
+        int count = 0;
+        for (int i = 0; i < field.length; i++) {
+            if (field[i][field.length - i - 1] == currentplayer
+                    || field[i + 1][field.length - i - 1] == currentplayer
+                    || field[i + 2][field.length - i - 1] == currentplayer
+                    || field[i][field.length - (i + 1) - 1] == currentplayer
+                    || field[i][field.length - (i + 2) - 1] == currentplayer
+                    || field[i][field.length - (i + 3) - 1] == currentplayer) {
+                count++;
+            } else {
+                count = 0;
+            }
+            if (count == 4) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public static boolean detectWinRow(int[][] field, int row, int currentplayer) {         //do i need a int row (from "i") or could i just use field.length ?
+        int count = 0;
+        for (int i = 0; i < field[row].length; i++) {
+            if (field[row][i] == currentplayer) {
+                count++;
+            } else {
+                count = 0;
+            }
+            if (count >= 4) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public static boolean detectWinCol(int[][] field, int col, int currentplayer) {
+        int count = 0;
+        for (int i = 0; i < field[col].length - 1; i++) {  //why field[col].length has to be -1???
```

Comment:
The reason why you need this - 1 is that you have mixed up rows and columns here.

field[x] refers to a col. field[x].length refers to length of a column. On your board, that is 7. In line 236 you use the column size to declare the max value of i. And then in line 237 you use i to index a row. Your row count is 6. => field[x].length is 7 but is used to index something which has max size 6. You make up for it by subtracting one. But the correct solution would be to set i = field.length aka. row count